platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommand.java

```diff
@@ -0,0 +1,202 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.command;
+
+import static org.boon.Boon.fromJson;
+import static org.fusesource.jansi.Ansi.ansi;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.karaf.features.Feature;
+import org.apache.karaf.shell.api.action.Option;
+import org.codice.ddf.admin.application.service.ApplicationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Extends the {@link AbstractApplicationCommand} with options and styling for profile subcommands
+ */
+public abstract class AbstractProfileCommand extends AbstractApplicationCommand {
+
+    @Option(name = "--dev", description = "Show development profiles", required = false)
+    boolean showDevProfiles = false;
+
+    Path developmentProfilePath = Paths.get(System.getProperty("ddf.home"), "etc", "admin", "profiles");
+
+    protected static final String DEV_PROFILE_INSTALL_FEATURES = "install-features";
+    protected static final String DEV_PROFILE_UNINSTALL_FEATURES = "uninstall-features";
+    protected static final String DEV_PROFILE_START_APPS = "start-apps";
+    protected static final String DEV_PROFILE_STOP_BUNDLES = "stop-bundles";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractProfileCommand.class);
+
+    /**
+     * Check if an installation profile exists
+     * @param installProfiles the list of installation profiles
+     * @param profileName the name of the profile to check for
+     * @return {@link Boolean} profile exists?
+     */
+    protected boolean profileExists(List<Feature> installProfiles, String profileName) {
+        return installProfiles.stream().anyMatch(feature -> feature.getName().contains(profileName));
+    }
+
+    /**
+     * Check if a dev profile exists (Experimental)
+     * @param profileName name of profile to check for
+     * @return {@link Boolean} profile exists?
+     */
+    protected boolean devProfileExists(String profileName) {
+        return Paths.get(developmentProfilePath.toString(), profileName + ".json").toFile().exists();
+    }
+
+    /**
+     * Get the {@link Feature} that makes up a profile
+     * @param applicationService {@link ApplicationService}
+     * @param profileName Name of profile to get
+     * @return {@link Feature}
+     */
+    protected Feature getProfile(ApplicationService applicationService, String profileName) {
+        return applicationService.getInstallationProfiles().stream()
+                .filter(i -> i.getName().contains(profileName))
+                .findFirst()
+                .get();
```

Comment:
❗️ Will throw an exception (I believe it's a `NoSuchElementException`) if no match is found. Change from `get` to `orElse` with a default return value (null would potentially be fine).